src/adagrams.js

```diff
@@ -1,8 +1,125 @@
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    const letterFreqs = {
+      A : 9,
+      B : 2,
+      C : 2,
+      D : 4,
+      E : 12,
+      F : 2,
+      G : 3, 
+      H : 2, 	
+      I : 9, 	
+      J : 1, 	
+      K : 1, 	
+      L : 4, 	
+      M : 2, 
+      N : 6, 
+	    O : 8, 
+	    P : 2, 
+	    Q : 1, 
+	    R : 6, 
+ 	    S : 4, 
+      T : 6, 
+      U : 4,
+      V : 2, 
+      W : 2,
+      X : 1,
+      Y : 2, 
+      Z : 1
+    };
+    const letters = Object.keys(letterFreqs);
+    const hand = [];
+    while(hand.length < 10){
+      const randLetter = letters[Math.floor(Math.random() * letters.length)];
+      if(letterFreqs[randLetter] > 0){
+        letterFreqs[randLetter] -= 1;
+        hand.push(randLetter);
+      } 
+    }
+    return hand;
   },
+
+  usesAvailableLetters(input, lettersInHand) {
+    const letterCount = {};
+
+    //refactor as forEach?
+    for(const letter of lettersInHand){
+      if(letterCount[letter]){
+        letterCount[letter] += 1;
+      }
+      else{
+        letterCount[letter] = 1;
+      }
+    }
+    
+    
+    for(const letter of input){
+      if(letterCount[letter] === 0 || letterCount[letter] === undefined){
+        return false;
+      }
+      else{
+        letterCount[letter] -= 1;
+      }
```

Comment:
`0` and are both falsey in JavaScript so you can write this as:

```suggestion
      if (letterCount[letter]) {
        letterCount[letter] -= 1;
      } else {
        return false;
      }
```

(I swapped the `if`/`else` branches because the `!` gets lost easily.)

Also, your spacing was a little funky.